lib/OpenQA/Worker/Job.pm

```diff
@@ -439,23 +439,27 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
-        {
-            next unless -e $file;
-
-            # replace some file names
-            my $ofile = $file;
-            $ofile =~ s/serial0/serial0.txt/;
-            $ofile =~ s/virtio_console.log/serial_terminal.txt/;
-
-            my %upload_parameter = (file => {file => "$pooldir/$file", filename => $ofile},);
-            if (!$self->_upload_log_file_or_asset(\%upload_parameter)) {
-                $reason = 'api-failure';
-                last;
+        if (defined $self->info->{settings}->{AUOTINST_UPLOAD_FROM_POOL}) {
+            my @files_to_upload = split(',', $self->info->{settings}->{AUOTINST_UPLOAD_FROM_POOL});
+            for my $file (@files_to_upload)
+            {
+                next unless -e $file;
+
+                # replace some file names
+                my $ofile = $file;
+                $ofile =~ s/serial0/serial0.txt/;
+                $ofile =~ s/virtio_console.log/serial_terminal.txt/;
+
+                my %upload_parameter = (file => {file => "$pooldir/$file", filename => $ofile},);
+                if (!$self->_upload_log_file_or_asset(\%upload_parameter)) {
+                    $reason = 'api-failure';
+                    last;
+                }
             }
         }
+        else {
+            log_error('AUOTINST_UPLOAD_FROM_POOL is not defined nothing will be uploaded from pool directory');
```

Comment:
Better put the old list in as default. Instead of if/else use `//`

Reply:
yeah agree with idea to fallback to old list , but let's do something in the middle . WDYT ? Still find log message needed

Reply:
and error in the system log of the worker? Who would look at this message?